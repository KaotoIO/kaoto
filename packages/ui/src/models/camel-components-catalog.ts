import { CatalogKind } from './catalog-kind';

export interface ICamelComponentDefinition {
  component: ICamelComponent;
  componentProperties: ICamelComponentProperties;
  headers: ICamelComponentHeaders;
  properties: ICamelComponentProperties;
}

export interface ICamelComponent {
  kind: CatalogKind.Component;
  name: string;
  title: string;
  description: string;
  deprecated: boolean;
  firstVersion?: string;
  label: string;
  javaType?: string;
  supportLevel?: string;
  groupId?: string;
  artifactId?: string;
  version: string;
  scheme?: string;
  extendsScheme?: string;
  syntax?: string;
  async?: boolean;
  api?: boolean;
  consumerOnly?: boolean;
  producerOnly?: boolean;
  lenientProperties?: boolean;
}

export interface CatalogItemCommon {
  index: number;
  kind: string;
  displayName: string;
  group: string;
  label: string;
  required: boolean;
  javaType: string;
  deprecated: boolean;
  deprecationNote: string;
  autowired: boolean;
  secret: boolean;
  description: string;
}

export interface ICamelComponentProperties {
  type: string;
}

export interface ICamelComponentHeaders extends CatalogItemCommon {
  constantName: string;
}
