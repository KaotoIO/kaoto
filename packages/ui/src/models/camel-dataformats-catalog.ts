import { ICamelProcessorModel, ICamelProcessorProperty } from './camel-processors-catalog';
import { KaotoSchemaDefinition } from './kaoto-schema';

export interface ICamelDataformatDefinition {
  model: ICamelDataformatModel;
  properties: Record<string, ICamelDataformatProperty>;
  propertiesSchema: KaotoSchemaDefinition['schema'];
}

export type ICamelDataformatModel = ICamelProcessorModel;

export type ICamelDataformatProperty = ICamelProcessorProperty;
