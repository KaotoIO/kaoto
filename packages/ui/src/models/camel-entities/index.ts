export * from './base-entity';

/** Camel Route definitions */
export * from './flows/route';
