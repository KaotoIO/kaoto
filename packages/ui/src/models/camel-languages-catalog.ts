import { JSONSchemaType } from 'ajv';
import { ICamelProcessorModel, ICamelProcessorProperty } from './camel-processors-catalog';

export interface ICamelLanguageDefinition {
  model: ICamelLanguageModel;
  properties: Record<string, ICamelLanguageProperty>;
  propertiesSchema: JSONSchemaType<unknown>;
}

export interface ICamelLanguageModel extends ICamelProcessorModel {}

export interface ICamelLanguageProperty extends ICamelProcessorProperty {}
