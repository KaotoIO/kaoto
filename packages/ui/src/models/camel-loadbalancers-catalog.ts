import { JSONSchemaType } from 'ajv';
import { ICamelProcessorModel, ICamelProcessorProperty } from './camel-processors-catalog';

export interface ICamelLoadBalancerDefinition {
  model: ICamelLoadBalancerModel;
  properties: Record<string, ICamelLoadBalancerProperty>;
  propertiesSchema: JSONSchemaType<unknown>;
}

export interface ICamelLoadBalancerModel extends ICamelProcessorModel {}

export interface ICamelLoadBalancerProperty extends ICamelProcessorProperty {}
