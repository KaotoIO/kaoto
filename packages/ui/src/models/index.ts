export * from './camel-catalog-index';
export * from './camel-components-catalog';
export * from './camel-processors-catalog';
export * from './catalog-kind';
export * from './flow';
export * from './kamelets-catalog';
export * from './schema';
