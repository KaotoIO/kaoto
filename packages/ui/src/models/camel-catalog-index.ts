import { ICamelComponentDefinition } from './camel-components-catalog';
import { ICamelProcessorDefinition } from './camel-processors-catalog';
import { CatalogKind } from './catalog-kind';
import { IKameletDefinition } from './kamelets-catalog';

export interface CamelCatalogIndex {
  catalogs: Catalogs;
  schemas: CatalogEntry[];
}

export interface Catalogs {
  models: CatalogEntry;
  components: CatalogEntry;
  languages: CatalogEntry;
  dataformats: CatalogEntry;
  kamelets: CatalogEntry;
}

export interface CatalogEntry {
  name: string;
  version: string;
  file: string;
}

export type CatalogTypes = Record<string, ICamelComponentDefinition | ICamelProcessorDefinition | IKameletDefinition>;

export interface ComponentsCatalog {
  [CatalogKind.Component]?: Record<string, ICamelComponentDefinition>;
  [CatalogKind.Processor]?: Record<string, ICamelProcessorDefinition>;
  [CatalogKind.Kamelet]?: Record<string, IKameletDefinition>;
  [key: string]: unknown;
}

export interface CatalogCamelComponent {
  type: CatalogKind.Component;
  definition: ICamelComponentDefinition;
}

export interface CatalogCamelProcessor {
  type: CatalogKind.Processor;
  definition: ICamelProcessorDefinition;
}

export interface CatalogKamelet {
  type: CatalogKind.Kamelet;
  definition: IKameletDefinition;
}
