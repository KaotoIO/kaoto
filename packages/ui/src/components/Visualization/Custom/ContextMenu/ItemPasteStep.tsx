import { PasteIcon } from '@patternfly/react-icons';
import { ContextMenuItem } from '@patternfly/react-topology';
import { FunctionComponent, PropsWithChildren } from 'react';
import { IDataTestID } from '../../../../models';
import { AddStepMode, IVisualizationNode } from '../../../../models/visualization/base-visual-entity';
import { usePasteStep } from '../hooks/paste-step.hook';

interface ItemPasteStepProps extends PropsWithChildren<IDataTestID> {
  vizNode: IVisualizationNode;
  mode: AddStepMode;
  text: string;
}

export const ItemPasteStep: FunctionComponent<ItemPasteStepProps> = (props) => {
  const { onPasteStep, isCompatible } = usePasteStep(props.vizNode, props.mode);

  if (!isCompatible) return null;

  return (
    <ContextMenuItem onClick={onPasteStep} data-testid={props['data-testid']}>
      <PasteIcon /> {props.text}
    </ContextMenuItem>
  );
};
