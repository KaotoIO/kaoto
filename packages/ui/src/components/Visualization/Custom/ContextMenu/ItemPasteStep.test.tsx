import { fireEvent, render } from '@testing-library/react';
import { AddStepMode, createVisualizationNode } from '../../../../models';
import { ItemPasteStep } from './ItemPasteStep';
import { usePasteStep } from '../hooks/paste-step.hook';

// Mock the `usePasteStep` hook
jest.mock('../hooks/paste-step.hook', () => ({
  usePasteStep: jest.fn(),
}));

describe('ItemPasteStep', () => {
  const vizNode = createVisualizationNode('test', {});
  const mockOnPasteStep = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render Paste ContextMenuItem', () => {
    // Mock the `usePasteStep` hook to return compatible state
    (usePasteStep as jest.Mock).mockReturnValue({
      onPasteStep: mockOnPasteStep,
      isCompatible: true,
    });

    const { container } = render(
      <ItemPasteStep vizNode={vizNode} mode={AddStepMode.AppendStep} text="Paste as child" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should not render Paste ContextMenuItem', () => {
    // Mock the `usePasteStep` hook to return compatible state
    (usePasteStep as jest.Mock).mockReturnValue({
      onPasteStep: mockOnPasteStep,
      isCompatible: false,
    });

    const { container } = render(
      <ItemPasteStep vizNode={vizNode} mode={AddStepMode.AppendStep} text="Paste as child" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should call onPasteStep when the context menu item is clicked', () => {
    // Mock the `usePasteStep` hook to return compatible state
    (usePasteStep as jest.Mock).mockReturnValue({
      onPasteStep: mockOnPasteStep,
      isCompatible: true,
    });

    const wrapper = render(
      <ItemPasteStep vizNode={vizNode} mode={AddStepMode.InsertChildStep} text="Paste as child" />,
    );
    fireEvent.click(wrapper.getByText('Paste as child'));

    expect(mockOnPasteStep).toHaveBeenCalledTimes(1);
  });
});
