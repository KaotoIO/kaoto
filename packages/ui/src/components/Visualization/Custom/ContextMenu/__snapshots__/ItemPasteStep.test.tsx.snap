// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCopyStep should not render Paste ContextMenuItem 1`] = `<div />`;

exports[`ItemCopyStep should render Paste ContextMenuItem 1`] = `
<div>
  <li
    class="pf-v6-c-menu__list-item"
    data-ouia-component-id="OUIA-Generated-DropdownItem-1"
    data-ouia-component-type="PF6/DropdownItem"
    data-ouia-safe="true"
    role="none"
  >
    <button
      class="pf-v6-c-menu__item"
      role="menuitem"
      tabindex="-1"
      type="button"
    >
      <span
        class="pf-v6-c-menu__item-main"
      >
        <span
          class="pf-v6-c-menu__item-text"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M128 184c0-30.879 25.122-56 56-56h136V56c0-13.255-10.745-24-24-24h-80.61C204.306 12.89 183.637 0 160 0s-44.306 12.89-55.39 32H24C10.745 32 0 42.745 0 56v336c0 13.255 10.745 24 24 24h104V184zm32-144c13.255 0 24 10.745 24 24s-10.745 24-24 24-24-10.745-24-24 10.745-24 24-24zm184 248h104v200c0 13.255-10.745 24-24 24H184c-13.255 0-24-10.745-24-24V184c0-13.255 10.745-24 24-24h136v104c0 13.2 10.8 24 24 24zm104-38.059V256h-96v-96h6.059a24 24 0 0 1 16.97 7.029l65.941 65.941a24.002 24.002 0 0 1 7.03 16.971z"
            />
          </svg>
           
          Paste as child
        </span>
      </span>
    </button>
  </li>
</div>
`;
